src/modules/Notifies/NotifyExtension.cpp

```diff
@@ -0,0 +1,138 @@
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Lesser General Public License for more details.
+
+	You should have received a copy of the GNU Lesser General Public License
+	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <NotifyExtension.hpp>
+
+#ifdef Q_OS_LINUX
+	#include <X11Notify.hpp>
+#endif
+#include <TrayNotify.hpp>
+
+#include <QCoreApplication>
+
+NotifyService::NotifyService(Notify *notify, Settings &settings) :
+	m_notify(notify),
+	m_lastPlayState(Stopped)
+{
+	Q_ASSERT(notify);
+	if (settings.getBool("ShowTitle"))
+		connect(&QMPlay2Core, SIGNAL(updatePlaying(bool, const QString &, const QString &, const QString &, int, bool, const QString &)), this, SLOT(updatePlaying(bool, const QString &, const QString &, const QString &, int, bool, const QString &)));
+	if (settings.getBool("ShowPlayState"))
+		connect(&QMPlay2Core, SIGNAL(playStateChanged(const QString &)), this, SLOT(playStateChanged(const QString &)));
+	if (settings.getBool("ShowVolume"))
+		connect(&QMPlay2Core, SIGNAL(volumeChanged(double)), this, SLOT(volumeChanged(double)));
+
+	if (settings.getBool("CustomMsg"))
+	{
+		m_summaryFormat = settings.getString("CustomSummary");
+		m_bodyFormat = settings.getString("CustomBody");
+	}
+}
+
+NotifyService::~NotifyService()
+{
+	delete m_notify;
+}
+
+void NotifyService::volumeChanged(double v)
+{
+	m_notify->showMessage(tr("Volume Changed"), tr("Volume: %1%").arg((int)(100 * v)));
+}
+
+void NotifyService::updatePlaying(bool play, const QString &title, const QString &artist, const QString &album, int, bool, const QString &fileName)
+{
+	if (!play)
+		return;
+
+	QString summary = m_summaryFormat;
+	if (summary.isEmpty())
+	{
+		summary = title;
+		if (!title.isEmpty() && !artist.isEmpty())
+			summary += " - ";
+		summary += artist;
+	}
+	else
+		summary.replace("%title%", title).replace("%artist%", artist).replace("%album%", album).replace("%filename%", fileName);
+
+	if (summary.isEmpty())
+		summary = fileName;
+
+	QString body = m_bodyFormat;
+	if (!body.isEmpty())
+		body.replace("%title%", title).replace("%artist%", artist).replace("%album%", album).replace("%filename%", fileName);
+
+	if (body.isEmpty())
+		body = album;
+
+	m_notify->showMessage(summary, body);
+}
+
+void NotifyService::playStateChanged(const QString &playState)
+{
+	const PlayState last = m_lastPlayState;
+	if (playState == "Playing")
+		m_lastPlayState = Playing;
+	else if (playState == "Paused")
+		m_lastPlayState = Paused;
+	else
+		m_lastPlayState = Stopped;
+
+	/* In those cases we don't show notification:
+	 *  1. The last one is the same as the current one
+	 *  2. The current one is Playing and the last one wasn't Paused
+	 */
+	if (m_lastPlayState != last && (m_lastPlayState != Playing || last == Paused))
+		m_notify->showMessage(QCoreApplication::applicationName(), playState);
+}
+
+/**/
+
+NotifyExtension::NotifyExtension(Module &module) : m_notify(NULL)
```

Comment:
```c++
NotifyExtension::NotifyExtension(Module &module) :
	m_notify(NULL)
{
```

Also maybe rename to `m_notifyService` (because not the same as in class above)?
And no need to line break between functions (consistent to header, if you have a time you can check all code for this) :D